ddtrace/opentelemetry/span_test.go

```diff
@@ -22,63 +24,248 @@ import (
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/httpmem"
 )
 
-func TestSpanMethods(t *testing.T) {
-	testData := struct {
-		env, srv, oldOp, newOp string
-		tags                   [][]string // key - tags[0][0], value - tags[0][1]
-	}{
-		env:   "test_env",
-		srv:   "test_srv",
-		oldOp: "old_op",
-		newOp: "new_op",
-		tags:  [][]string{{"tag_1", "tag_1_val"}, {"opt_tag_1", "opt_tag_1_val"}}}
-	done := make(chan struct{})
-	var payload string
+func mockTracerProvider(t *testing.T, opts ...tracer.StartOption) (tp *TracerProvider, payloads chan string, cleanup func()) {
+	payloads = make(chan string)
 	s, c := httpmem.ServerAndClient(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		switch r.URL.Path {
 		case "/v0.4/traces":
+			if h := r.Header.Get("X-Datadog-Trace-Count"); h == "0" {
+				return
+			}
 			buf, err := io.ReadAll(r.Body)
-			if err != nil {
+			if err != nil || len(buf) == 0 {
+				t.Log("Test agent: Error receiving traces")
 				t.Fail()
 			}
-			payload = fmt.Sprintf("%s", buf)
-			done <- struct{}{}
+			var js bytes.Buffer
+			msgp.UnmarshalAsJSON(&js, buf)
+			payloads <- js.String()
 		}
 		w.WriteHeader(200)
 	}))
-	defer s.Close()
+	opts = append(opts, tracer.WithHTTPClient(c))
+	tp = NewTracerProvider(opts...)
+	otel.SetTracerProvider(tp)
+	return tp, payloads, func() {
+		tp.Shutdown()
+		s.Close()
+	}
+}
+
+func waitForPayload(ctx context.Context, payloads chan string) (string, error) {
+	var p string
+	select {
+	case <-ctx.Done():
+		return "", fmt.Errorf("Timed out waiting for traces")
+	case p = <-payloads:
+		break
+	}
+	return p, nil
+}
+
+func TestSpanSetName(t *testing.T) {
+	assert := assert.New(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
```suggestion
	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```
1 second should be more than enough

Reply:
(here and throughout)